File: src/main/java/com/quid/kopring/controller/user/UserController.java
package com.quid.kopring.controller.user;

import com.quid.kopring.user.model.request.UserCreateRequest;
import com.quid.kopring.user.model.request.UserUpdateRequest;
import com.quid.kopring.user.model.response.UserResponse;
import com.quid.kopring.user.service.UserService;
import java.util.List;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping("/user")
    public void saveUser(@RequestBody UserCreateRequest request) {
        userService.saveUser(request);
    }

    @GetMapping("/user")
    public List<UserResponse> getUsers() {
        return userService.getUsers();
    }

    @PutMapping("/user")
    public void updateUserName(@RequestBody UserUpdateRequest request) {
        userService.updateUserName(request);
    }

    @DeleteMapping("/user")
    public void deleteUser(@RequestParam String name) {
        userService.deleteUser(name);
    }

}

File: src/main/kotlin/com/quid/kopring/user/controller/UserController.kt
package com.quid.kopring.user.controller

import com.quid.kopring.user.model.request.UserCreateRequest
import com.quid.kopring.user.model.request.UserUpdateRequest
import com.quid.kopring.user.model.response.UserLoanListResponse
import com.quid.kopring.user.model.response.UserResponse
import com.quid.kopring.user.service.UserService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
class UserController (
    private val userService: UserService
){
    @PostMapping
    fun saveUser(@RequestBody request: UserCreateRequest) {
        userService.saveUser(request)
    }

    @GetMapping
    fun getUsers(): List<UserResponse> = userService.getUsers()

    @PutMapping
    fun updateUser(@RequestBody request: UserUpdateRequest) {
        userService.updateUserName(request)
    }

    @DeleteMapping
    fun deleteUser(@RequestParam request: String) {
        userService.deleteUser(request)
    }

    @GetMapping("/loanList")
    fun getUserLoanList(@RequestParam name: String): UserLoanListResponse {
        return userService.getUserLoanList(name)
    }
}